pkg/orchestrator/gitHubActions.go

```diff
@@ -103,3 +184,33 @@ func isGitHubActions() bool {
 	envVars := []string{"GITHUB_ACTION", "GITHUB_ACTIONS"}
 	return areIndicatingEnvVarsSet(envVars)
 }
+
+func (g *GitHubActionsConfigProvider) GetStageIds() ([]int, error) {
+	resp, err := g.client.GetRequest(fmt.Sprintf("%s/runs/%s/jobs", getActionsURL(), getEnv("GITHUB_RUN_ID", "")), g.getHeader(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get API data: %w", err)
+	}
+
+	var stagesID StagesID
+	err = piperHttp.ParseHTTPResponseBodyJSON(resp, &stagesID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JSON data: %w", err)
+	}
+
+	ids := make([]int, len(stagesID.Jobs))
+	for i, job := range stagesID.Jobs {
+		ids[i] = job.ID
+	}
+	if len(ids) == 0 {
+		return nil, fmt.Errorf("failed to get IDs")
+	}
+
+	// execution of the last stage hasn't finished yet - we can't get logs of the last stage
+	return ids[:len(stagesID.Jobs)-1], nil
+}
+
+func (g *GitHubActionsConfigProvider) getHeader() http.Header {
```

Comment:
Since the HTTP header is needed in a couple of places and probably will be needed for methods that are not implemented yet we decided to use the method, which returns the HTTP header, to avoid duplication. Previously the method also retrieved GitHub token. 
I think it is better to use constant instead of the method now.

Reply:
Ok. I suggest adding it to the description comment I added as a placeholder. 
However, reading your answer, didn't get for what it is really needed.

Reply:
Previously it returned the header with a GitHub token, but now, since we retrieve token another way, this method is not needed anymore. I deleted it. [6f074e2](https://github.com/SAP/jenkins-library/pull/4250/commits/6f074e285404afc617d0efdb5a739801b536a668)